### DISCUSSION TITLE
OIDC ID-token sign with ES256/ES512 throw me error ( type assert key to *rsa.PublicKey)

### DISCUSSION THREAD

**Discussion by epicaqua** [ory/hydra#3106] [Q&A] [answered]
Dear team,

I was trying to change id-token sign alg from RS256 to ES256  'hydra.openid.id-token', 

I followed the key rotation steps on [link](https://www.ory.sh/docs/hydra/guides/secrets-key-rotation#rotation-of-json-web-token-signing-keys ) to generate the key and was able to see it in API call `GET /keys/hydra.openid.id-token.`

However I get below error when requesting token, also when I restarted hydra after addition of ES256/512 key it gave the same error and wouldn't start.  

Error log:
`error=map[message:unable to type assert key to *rsa.PublicKey] service_name=Ory Hydra service_version=v1.11.8`

1. Is there a way to change it from some configs or something? 

2. is it supported already? 

I found something closed discussion related to it [ISSUE: 1638](https://github.com/ory/hydra/issues/1638)  also there are other jwt strategies in code but not sure if they are only related to client authentication.

**Comment by vinckr** [comment, marked as answer]
Hello @epicaqua,
This error is going to be fixed as a part of Ory Hydra v2.0 (https://github.com/ory/hydra/pull/2796).

**Reply by epicaqua** [reply]
thanks for info
